src/modules/components/proposal/proposalDataListItem/proposalDataListItemResult/proposalDataListItemResult.tsx

```diff
@@ -0,0 +1,53 @@
+import { Progress, formatterUtils } from '../../../../../core';
+import { type IApprovalThresholdResult, type IMajorityVotingResult } from '../proposalDataListItemStructure';
+
+export type IProposalDataListItemResultProps = IApprovalThresholdResult | IMajorityVotingResult;
+
+/**
+ * `ProposalDataListItemResult` component
+ */
+export const ProposalDataListItemResult: React.FC<IProposalDataListItemResultProps> = (props) => {
```

Comment:
Ngl, these two returns inside if statements looks pretty ugly 😅 On one end we could split this into two components, this will make the component easier to test and more scalable when adding more result types. On the other end it is pretty small right now and adding more folders/components might be overkill :/